gotchas/styled-jsx-in-ui-core.md

```diff
@@ -0,0 +1,9 @@
+# Usage of styled-jsx in our ui-core library
+
+With styled-jsx, for our components in ui-core:
+
+- We try to target the element (e.g. https://github.com/dhis2/ui-core/blob/master/src/Button/styles.js#L6)
```

Comment:
We should set up some tests with pros and cons, I recently did some experimenting with this and realised that I was slightly mistaken in some of the side-effects, so let's get to the root of this. Right now, I am not confident in saying that targeting the element itself is always the best option.

Reply:
I thought that using class based selectors increases the chance of global styles bleeding in, and that's why we try to avoid them if possible. However, since we are using the `classnames` package all over the place, the risk of global styles bleeding in is very real anyway. So we have just accepted that and documented that. As such we could argue that the advantage of targeting elements is not really that relevant.

Perhaps we should just say something like:
- If the component only returns a simple JSX structure, with a single occurrence of each tag it contains, target elements.
- Otherwise, just go for class-name based styles